.github/workflows/container-structure-test.yml

```diff
@@ -12,13 +14,38 @@ jobs:
     steps:
       - name: checkout source
         uses: actions/checkout@master
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
       - name: Install Google Container Structure Test Framework
         run: curl -LO https://storage.googleapis.com/container-structure-test/latest/container-structure-test-linux-amd64 && chmod +x container-structure-test-linux-amd64 && sudo mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
-      - name: Build docker images PHP Only
-        run: docker build -t antistatique/php-dev:${{ matrix.php_version }} -f ./php/${{ matrix.php_version }}/Dockerfile .
-      - name: Run tests PHP only
-        run: container-structure-test test --image antistatique/php-dev:${{ matrix.php_version }} --config ./tests/baseMetadataTests.yaml --config ./tests/baseFileExistenceTests.yaml --config ./php/${{ matrix.php_version }}/tests/config--${{ matrix.php_version }}.yaml
-
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
```

Comment:
can we move the login step before the push step or this is a requirement for `docker-build-push-actions` ?

Reply:
ok make sense to have it here, as the build pull images from docker hub.